ceph/ceph_admin/helper.py

```diff
@@ -763,6 +764,34 @@ def file_or_path_exists(node, file_or_path):
     return False
 
 
+def file_or_path_updated(node, file_or_path, interval=10):
+    """
+    Method to check if the given file or directory was updated in the given interval
+    Args:
+        node: node object where file should be exists
+        file_or_path: ceph file or directory path
+        interval: if mentioned checks whether the file existing in the given path
+                              was updated anytime between now and the interval given here.
+
+    Returns:
+        boolean
+    """
+    try:
+        out, _ = node.exec_command(cmd=f"date -u -r {file_or_path}", sudo=True)
+        LOG.info(f"Output : {out}")
+        modified_time = parser.parse(out).replace(tzinfo=None)
+        time_now = datetime.utcnow().replace(tzinfo=None)
+        time_diff = (time_now - modified_time).seconds
+        LOG.info(f"Time difference between now and gz file modified date : {time_diff}")
+        if time_diff > 120:
+            return False
+        return True
+    except CommandFailed as err:
+        LOG.error(f"Error: {err}")
```

Comment:
I see below issues using this try..catch block here -
1. `CommandFailed` exception handled here is specific to only one statement i.e for `node.exec_command`. This makes no effect on adding other statement inside `try..catch` block.
2. In case of command failure we adding simple error log and returning `False` which is checked at calling time and again returning `False` with simple error message. In this case it will be harder to get stacktrace in case of TC failure.
3. [1] using this module will be more reliable than executing command.

[1] https://docs.paramiko.org/en/stable/api/sftp.html#paramiko.sftp_client.SFTPClient.lstat

Reply:
I have added only the required statement inside the try catch block. The paramiko implementation, as discussed let us take it up as a different task since it is out of scope for the current PR.